### ISSUE TITLE
About dialog can go offscreen vertically

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#516]
From https://github.com/phetsims/resistance-in-a-wire/issues/172, we found that when using stringTest=double, the credits patterns are doubled, and hence so are the credit strings, such that there is enough line wrap to kick the top of the about dialog off the screen. 

At first thought I wonder if Dialog should be able to handle this gracefully with a maxHeight value and scaling but I don't see and maxHeight support in Dialog.  Also note that this issue may belong in sun (for Dialog) rather than joist (for AboutDialog).

**Comment by zepumph** [comment]
@ariel-phet please assign and prioritize.

**Comment by ariel-phet** [comment]
marking for a dev meeting (quick) discussion. We might punt on this issue until the first sim that comes along with a big about dialog (and that developer will be the lucky winner!)

**Comment by pixelzoom** [comment]
AboutDialog currently applies `var MAX_WIDTH = 550` to each of its text element individually, so that they aren't all impacted by one especially-long string.  

AboutDialog extends Dialog extends Panel extends, and `options` are propagated all the way through. So you should be able to set `maxHeight` for AboutDialog to resolve this issue.

But I'm wondering if maxWidth and maxHeight should be percentages of layoutBounds, rather than constants.

**Comment by pixelzoom** [comment]
Here's RIAW (master) with `?stringTest=long&brand=phet&dev`.  Not quite outside the layoutBounds, but close.

![screenshot_750](https://user-images.githubusercontent.com/3046552/45126396-6e303600-b130-11e8-8b89-236797fb87e2.png)

Here's what it looks like if AboutDialog does `Dialog.call( this, { ..., maxHeight: 375 } )`:

![screenshot_751](https://user-images.githubusercontent.com/3046552/45126531-03cbc580-b131-11e8-8685-6a9ac614fcfe.png)

Also note that the AboutDialog is not centered in the ScreenView layoutBounds, it's centered in an area that takes the navbar height into account.  Perhaps that should be changed.

**Comment by samreid** [comment]
It seems more aesthetically pleasing for it to be centered in the ScreenView (doesn't include the Navigation Bar).  `maxHeight` seems to be a reasonable and easy solution.

**Comment by samreid** [comment]
For testing, I added these lines to resistance in a wire credits and tested with ?stringTest=long
```
      contributors: 'Wendy Adams, Mindy Gratny, Ariel Paul',
      graphicArts: 'Wendy Adams, Mindy Gratny, Ariel Paul',
      soundDesign: 'Wendy Adams, Mindy Gratny, Ariel Paul',
      translation: 'Wendy Adams, Mindy Gratny, Ariel Paul',
```

Commit coming up.

**Comment by samreid** [comment]
Proposed fix is committed, notes are in the comments in the commit.  For now, I centered the AboutDialog above the navigation bar, but if we decide to center all Dialogs above the navigation bar, we could move this layoutStrategy from AboutDialog to `Dialog.DEFAULT_LAYOUT_STRATEGY`.  @ariel-phet can you please assign a developer to review/test?

**Comment by ariel-phet** [comment]
@Denz1994 said he could take a look, seems like it should be pretty quick. If all looks good, feel free to close @Denz1994

**Comment by Denz1994** [comment]
I tested locally, with John Travoltage and MAS. The about dialog remains centered at various screen sizes and works well with `stringTest=long` and `stringTest=x`. The documentation added to the commited changes is descriptive aswell. This looks good as is @samreid.

**Comment by Denz1994** [comment]
Making the screen vertically narrow causes the dialog box layer to become horizontally narrower than the screen bounds. The side effect is that the nav-bar becomes pickable, which goes against the modal nature of the about dialog.

I tested with MAS and John-Travoltage. This doesn't effect keyboard nav as the nav-bar remains unfocusable while the about dialog is open, but is still pickable, so the "PhET" combo box can be opened along with the keyboard nav box and an about dialog at the same time as seen in the below image. Chorme has some vertical browser restrictions that prevent this level of narrowness, but I was able to reproduce on Firefox.

![image](https://user-images.githubusercontent.com/13894380/48870247-0fcd3800-edae-11e8-9fa5-10990224bbf5.png)

I don't think this is related to @samreid commits, but perhaps to the `this.modalNodeStack` which is being used to initialize `this.barrierRectangle` as a child of `this.topLayer` in `JOIST/Sim.showPopup()` 

Because the screen has to be made so narrow I don't see this as a current issue, but I thought I should document for potential future cases. Thoughts @samreid?

**Comment by jonathanolson** [comment]
> The long term solution would be for Plane to use infinite bounds. @jonathanolson can you comment on whether that is feasible?

Yes, should be feasible, and that is the plan. Just hasn't been a priority.

**Comment by KatieWoe** [comment]
Noticed on https://github.com/phetsims/QA/issues/487, indicating this is still an issue.
![abouttoobig](https://user-images.githubusercontent.com/41024075/77480254-f5aeca00-6de6-11ea-87ff-2758dd7003ed.png)

**Comment by JRomero0613** [comment]
This does not occur for my own Windows 10 device nor the iOS 13.3.1 iPad I am testing on. It might have something to do with the size of my screens because both devices have plenty of room vertically for the about dialog.
